README.md

```diff
@@ -1,39 +1,91 @@
 # docker-ops
 
-Bash scripts to release docker images.
+Bash functions to support release cycle of docker images.
 
-## Install
-All the docker-ops functions are available after you install it.
+## Why and features
 
-### In current shell
-If you want to run docker-ops in current shell:
+ * `docker-ops` is a single file, versioned and released after testing the functions
+ * Any project can quickly and easily reference the functions by downloading a specific release from github
+ * The `docker-ops` file is easy to review and hack with, there is no magic framework to learn, just plain bash and docker commands. When something does not fit your needs, it is easy to fall back to writing your own script.
+ * It works on nearly any setup because of minimal dependencies
+
+Use the functions as you like, but the primary goal is to support following **lifecycle of docker images**:
+1. Build image from source code and tag it with the git sha256 of the origin commit. Push this image to docker registry. We call it the *temp* image.
+   * Supporting function is `docker_ops::docker_build`.
+   * `docker_ops::docker_build` produces 3 [`imagerc` files](#imagerc-files), each has the same data in different format (JSON, YAML, bash). It contains critical information about the built image - the docker registry, image name, image tag.
+2. Pull the *temp* image (if not available locally) with `docker_ops::ensure_pulled_image`. Test the image to ensure desired level of quality. Use the right tools for the job, `docker-ops` has nothing to do with this step.
```

Comment:
I'd add info when/why this step is necessary. It is necessary to run on CI agents, because one agent might have built the docker image and another agent will be responsible for testing the docker image.

Also I would replace: "`docker-ops` has nothing to do with this step" with "`docker-ops` has nothing to do with docker image testing".